server/graphql-schema.js

```diff
@@ -65,40 +65,108 @@ const graphqlSchema = gql`
+        latestVersionWithResults: TestPlanVersion
+        versions: [TestPlanVersion]!
+        versionsWithResults: [TestPlanVersion]!
+        queueRelevantVersions: [TestPlanVersion]!
+    }
+
+    type TestPlanVersion {
         title: String!
-        publishStatus: TestPlanStatus!
-        revision: String!
-        sourceGitCommit: String!
+        status: TestPlanStatus!
+        gitSha: String!
+        gitMessage: String!
         exampleUrl: String!
-        # TODO: consider renaming createdAt and including the date the source
-        # code age was authored as well.
-        createdAt: Timestamp!
+        updatedAt: Timestamp!
         tests: [Test]!
         testCount: Int!
+        testPlanReports: [TestPlanReport]!
+        testPlanReport(id: ID, testPlanTarget: ID): TestPlanReport
     }
 
     interface Test {
         title: String!
-        totalAssertions: Int!
-        # TODO: need complete representation of test data
+        index: Int!
+        # TODO: account for running scripts
+        instructions: [Instruction]!
+        assertions: [Assertion]!
+        assertionCount: Int!
+        optionalAssertionCount: Int!
+        passThroughs: [PassThrough]!
+    }
+
+    type Instruction {
+        # TODO: account for automation
+        manualInstruction: String!
+    }
+
+    interface Assertion {
+        # TODO: account for at-specific assertions
+        # TODO: account for optional assertions
+        # TODO: account for automation
+        manualAssertion: String!
+    }
+
+    interface PassThrough {
+        atMode: String!
+        nthCommand: Int!
+        # Examples would go here if we support multiple examples for one test.
     }
 
     type TestResult implements Test {
         title: String!
+        index: Int!
+        instructions: [Instruction]!
+        assertions: [Assertion]!
+        passThroughs: [PassThrough]!
+        assertionCount: Int!
+        optionalAssertionCount: Int!
+
         startedAt: Timestamp!
-        completedAt: Timestamp
-        testPlanRun: TestPlanRun
-        passedAssertions: Int
-        totalAssertions: Int!
-        # TODO: need complete representation of test data & results
+        completedAt: Timestamp!
+        isComplete: Boolean!
+        isSkipped: Boolean!
+
+        passThroughResults: [PassThroughResult]!
+        assertionsPassed: Int!
+        optionalAssertionsPassed: Int!
+        unexpectedBehaviorCount: Int!
+    }
+
+    type PassThroughResult implements PassThrough {
+        atMode: String!
+        nthCommand: Int!
+
+        output: String!
+        assertionResults: [AssertionResult]!
+        unexpectedBehaviors: [UnexpectedBehavior]!
+    }
+
+    type AssertionResult implements Assertion {
+        manualAssertion: String!
+        passed: Boolean!
+    }
+
+    type UnexpectedBehavior {
+        id: ID!
+        description: String!
+    }
+
+    type TestResultConflict {
+        breadcrumbs: String! # TBD
```

Comment:
It's actually pretty common to need to refer / point at deeply nested places in the graph, so I was thinking about introducing a type that would allow that. It's a way to say, "I want you to look at this test plan, this version, this report and this run."

It's not actually a string, that's just a placeholder.

My current best name for it is "Traversal" and it's starting to take form in my current branch. But definitely still very TBD. I can show more on Monday ... assuming I don't decide it's all a horrible idea, haha.

Reply:
Ok I follow you. Great idea there
And I hope it's the former for Monday, rather than the latter 😄